bestbook/views/__init__.py

```diff
@@ -115,17 +108,36 @@ class Login(MethodView):
     def post(self):
         email = request.form.get("email")
         password = request.form.get("password")
-        return jsonify(login(email, password))
+        result = login(email, password)
+        return result
 
 class Observe(MethodView):
     def post(self):
         observation = request.form
         return jsonify(observation)
 
 class Submit(MethodView):
-    def post(self):
-        pass    
 
+    @rest
+    def post(self):
+        topic = request.form.get('topic')
+        winner = request.form.get('winner')
+        candidates = request.form.get('candidates')
+        description = request.get('description')
+        source = request.get('source')
+        username = session.get('username')
+        if source:
+            description += " (%s)" % source
+
+        if not username:
+            raise Exception('Login required')
+
+        rec = Recommendation.add(
+            topic, winner,
+            [Book.clean_olid(c) for c in candidates.split(' ')],
+            username, description)
+        return rec.__dict__
```

Comment:
this line doesn't work :) json serialization is messing up due to datetime in the obj for `created` field.

Tried jsonify (removing `@rest`), tried json.dumps w/ alternate encoder

Reply:
https://stackoverflow.com/questions/11875770/how-to-overcome-datetime-datetime-not-json-serializable